### ISSUE TITLE
translate attributes only for element with whitespace "content"

### ISSUE THREAD

**Issue by DatenMetzgerX** [angular-translate/angular-translate#1338]
I created a little [jsfiddle](https://jsfiddle.net/szqqLq6q/3/) that shows the issue. Angular translate removes the icon from the link as it expects that it should translate the content of the link. But that's not the case, it should only translate the attribute. 

The issue disappears if the content is not formatted, e.g. when we define the link like this:

```html
<a href="#" translate translate-attr-title="Home"><i class="fa fa-home"></i></a>
```

the point is, that the text of an element with nicely formatted child elements is not undefined but an empty string. 

Solution. Do no test if the translation id is undefined but test if the translation id is falsy. 
I expect that we nee to adjust the condition on line 245 of the translate directive.

If you agree with the provided solution, then I will prepare a pull request that adjust the behaviour.

**Comment by knalli** [comment]
A proper solution would be a split directive.

I don't think the falsy approuch would work. Can you explain a bit more?

**Comment by DatenMetzgerX** [comment]
Yes, having two different directives would be the cleaner solution. But as this would be a major (breaking) change, so I don't think we can consider this as a solution. 

The implementation checks in the `updateTranslations` functions if the translation id for a specific key (i think this is the attribute/element-name) is defined. Why can we not change the check to exclude everything where the check is falsy.

before:
```js
var updateTranslations = function () {
     for (var key in translationIds) {
         if (translationIds.hasOwnProperty(key) && translationIds[key] !== undefined) {
             updateTranslation(key, translationIds[key], scope, scope.interpolateParams, scope.defaultText);
        }
    }
};
```


after:
```js
var updateTranslations = function () {
     for (var key in translationIds) {
         if (translationIds.hasOwnProperty(key) && !translationIds[key]) {
             updateTranslation(key, translationIds[key], scope, scope.interpolateParams, scope.defaultText);
        }
    }
};
```

From my point of view I believe that translating "", null or undefined is kind of useless. The translation id should always be a string (`attr` and also `text` return a string or undefined). 

Or do I miss a point?

**Comment by knalli** [comment]
Solved by your PR #1339
